### PR TITLE
ui: improve tabbar styling

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

The pull-request fixes some user-experience issues related to the tabbar styling.
Previously, when cycling through the tabs a shift would occur with the tab, file-icons, and close icons.

The pull-request addresses these problems by:
- adding a "transparent" border for all incative tabs so when a tab is activated and an active border is shown they will not shift compared to inactive tabs.
- removes extraneous code which caused file-icons and the close icon to shift.

https://user-images.githubusercontent.com/40359487/156239318-be75b948-5365-498b-825d-85c399200853.mp4

#### How to test

1. start the application
2. open multiple editors
3. confirm that switching between editors yields a consistent experience (no shifting of elements) compared to master
4. confirm that tabs in the bottom area also behave well

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/core/src/browser/style/tabs.css [hunk 1]
```diff
@@ -261,22 +262,14 @@ body.theia-editor-highlightModifiedTabs
   padding-right: 4px;
 }
 
-.p-TabBar.theia-app-centers .p-TabBar-tab.p-mod-current > .theia-tab-icon-label > .p-TabBar-tabIcon {
-  margin-top: -2px;
-}
-
-.p-TabBar.theia-app-centers .p-TabBar-tab.p-mod-current > .p-TabBar-tabCloseIcon {
-  margin-top: 0px;
-}
```
Comment (vince-fugnitto) [inline packages/core/src/browser/style/tabs.css:270]: @msujew I'm not sure why the changes were necessary for the current tab but it caused the shifting of the `file-icon` and the `close` to occur:

master-tab

Comment (msujew) [reply packages/core/src/browser/style/tabs.css:270]: Interesting, that only seems to happen on Chrome. I tested this (and build the original PR) on Firefox, where these lines of CSS fixed the shifting issue. But it seems like they introduced shifting in Chrome

**Review by msujew — APPROVED** [review]
Looks good to me 👍

**Comment by vince-fugnitto** [comment]
I noticed a small regression with the preferences-ui which I'll address (unnecessary top border):

image
